force-app/main/default/lwc/recurringDonationTable/recurringDonationTable.html

```diff
@@ -5,7 +5,11 @@
                 data={data}
                 hide-checkbox-column="true"
                 class="slds-max-medium-table_stacked"
-                columns={columns}>
+                columns={columns}
+                onrowaction={handleRowAction}>
         </lightning-datatable>
     </div>
+    <c-update-payment-method-modal></c-update-payment-method-modal>
```

Comment:
An alternative approach to open/close would be to provide an @api accessible variable on each of these components, then set those variables to true/false based on what is pressed. That would avoid the `querySelector` and repeated `openmodal`, `closeModal` methods.

Reply:
good suggestion! changed